website/server/libs/cron.js

```diff
@@ -66,20 +66,37 @@ function grantEndOfTheMonthPerks (user, now) {
 
   if (elapsedMonths > 0) {
     plan.dateUpdated = now;
-    // For every month, inc their "consecutive months" counter. Give perks based on consecutive blocks
-    // If they already got perks for those blocks (eg, 6mo subscription, subscription gifts, etc) - then dec the offset until it hits 0
-    _.defaults(plan.consecutive, {count: 0, offset: 0, trinkets: 0, gemCapExtra: 0});
-
-    for (let i = 0; i < elapsedMonths; i++) {
-      plan.consecutive.count++;
-
-      if (plan.consecutive.offset > 1) {
-        plan.consecutive.offset--;
-      } else if (plan.consecutive.count % 3 === 0) { // every 3 months
-        if (plan.consecutive.offset === 1) plan.consecutive.offset--;
-        plan.consecutive.trinkets++;
-        plan.consecutive.gemCapExtra += 5;
-        if (plan.consecutive.gemCapExtra > 25) plan.consecutive.gemCapExtra = 25; // cap it at 50 (hard 25 limit + extra 25)
+
+    let dateCreatedMoment = moment(plan.dateCreated).startOf('month');
```

Comment:
I'm afraid you can't rely on plan.dateCreated to be the start date of the subscription that the user currently has. If they had a previous subscription (e.g., a one-month recurring subscription) and they changed it to a different kind of subscription (e.g., six-month recurring), then dateCreated will be the date their first subscription was created. It can't be used to get an accurate calculation for whether the six-month period is finished.